images/airflow/2.9.2/python/mwaa/entrypoint.py

```diff
@@ -602,13 +602,18 @@ def run_airflow_command(cmd: str, environ: Dict[str, str]):
             subprocesses = _create_airflow_scheduler_subprocesses(environ, conditions)
             # Schedulers, triggers, and DAG processors are all essential processes and
             # if any fails, we want to exit the container and let it restart.
-            run_subprocesses(subprocesses, essential_subprocesses=subprocesses)
+            run_subprocesses(
+                subprocesses=[],
```

Comment:
There is the health check that looks for running AIrflow processes, how was this presenting itself in that loop?

Should that loop verify that only 1 Airflow process is present?

Reply:
The healthcheck that does this is the ECS healthcheck. If failed, this healthcheck will bounce the containers. So I am not sure if this would be the best mechanism to detect an issue like this, since bouncing the containers wouldn't fix the issue and just create further interruptions / issues. 

I think we should invest in some "invasive" testing in our image build pipeline against a running container to catch issues like these.